skargo/src/manifest.sk

```diff
@@ -1,43 +1,28 @@
 module Skargo;
 
-class Version(major: Int, minor: Int, patch: Int) uses Orderable {
-  static fun fromString(str: String): Version {
-    components = str.split(".");
-    if (components.size() != 3) {
-      invariant_violation("Invalid version string " + str)
-    };
-    Version(components[0].toInt(), components[1].toInt(), components[2].toInt())
-  }
-
-  fun toString(): String {
-    `${this.major}.${this.minor}.${this.patch}`
-  }
-}
-
-base class VersionConstraint uses Equality {
-  children =
-  | RangeConstraint(low: Version, high: ?Version)
-  | PathConstraint(path: String, version: ?Version)
-
-  static fun fromTOML(basePath: String, toml: TOML.Value): VersionConstraint {
+// TODO: Add support for `git` repositories as a package location.
+class ManifestDependency{
+  version: ?Semver.VersionReq = None(),
+  path: ?String = None(),
+} {
+  static fun fromTOML(basePath: String, toml: TOML.Value): ManifestDependency {
     toml match {
-    | TOML.String(v) -> static::fromString(v)
+    | TOML.String(v) ->
```

Comment:
Don't you want to allow `{ version = ... }` as well?

Reply:
Yes, the current grammar for `Skargo.toml` is a bit limited.